.github/workflows/npm-publish.yml

```diff
@@ -67,3 +67,31 @@ jobs:
           body_path: CHANGELOG.md
         env:
           GITHUB_TOKEN: ${{ github.token }}
+
+      # Deploy storybook after Publish to NPM
```

Comment:
Why not a separate job with the same checks?

Reply:
Why not adjust the previous storybook action to also deploy on push to main? 🤔

Reply:
It seems that the `release` event that is triggered by `npm-publish.yml` is not creating a new workflow run.

From the documentation:
```
When you use the repository's GITHUB_TOKEN to perform tasks on behalf of the GitHub Actions app, events triggered by the GITHUB_TOKEN will not create a new workflow run.
This prevents you from accidentally creating recursive workflow runs.
For example, if a workflow run pushes code using the repository's GITHUB_TOKEN, a new workflow will not run even when the repository contains a workflow configured to run when push events occur.
```
I added the storybook deployment after `Publish to NPM` job in the `npm-publish.yml`

> Why not adjust the previous storybook action to also deploy on push to main?

Because we only want to deploy a new storybook only when we publish a new package.

Reply:
> Why not a separate job with the same checks?

Something like this?
```
jobs:
  extract_npm_version:
	[…]
  publish_to_npm:
    needs: extract_npm_version
	[…]
  deploy_storybook:
    needs: extract_npm_version
	[…]
```

Reply:
Docs says it does: https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#release

Reply:
@katspaugh the problem is that it doesn't work when they are triggering it from another action. It's a security measure from Github to avoid recursive actions.

The idea of doing it like this was to use the exact same codebase, yarn install and checks.This way it creates the npm package and get then builds the storybook to be published by devops. We only wan't to bundle a new storybook each time a npm, package is published so it makes sense

Reply:
Oh, I see.

Reply:
> Another reason for doing it on the same action it's because we are only providing devops tagged commits for deployments.

Why are we doing this instead of using tags?

I'm sorry, but this PR is missing a lot of things like what problem it's going to solve and how; it also changes conditions on when actions are run instead of just moving stuff from one action to another

Reply:
> Because we only want to deploy a new storybook only when we publish a new package.

Could you just define the same run condition for publishing a storybook & publishing a new package?

Reply:
That's what I would do as well. To not mix two different actions.

Reply:
The other option we have, that I wanted to avoid not to be waiting for two actions to run is to set the release as "draft" and wait until npm package is published, then the release is created and publish it manually.

The only thing I don't like to be waiting for actions to finish in order to complete the flow.

Ideally the tag should be in place when creating the build, that way we can ensure that we upload the required code for each tag to devops infrastructure

Reply:
ok, then I restored Github actions.  New releases are set as `draft` now.

Reply:
Draft releases aren't ideal either. We want to avoid manual actions.

Reply:
Can't the storybook action be triggered on a tag push?

Reply:
@katspaugh :laughing: no matter the event, a Github action can't trigger another action. Even the create tag event won't trigger the action automatically. We already tested that. And that's why we were trying to avoid the manual action adding storybook bundling into npm action

Reply:
Damn it :D
Why not just do the same checks version then? That would be also in parallel to npm publish.